lib/expression/eval.ex

```diff
@@ -215,10 +215,15 @@ defmodule Expression.Eval do
       end
 
     comparison_result =
-      if is_struct(date_a, Date) do
-        Date.compare(date_a, date_b)
-      else
-        DateTime.compare(date_a, date_b)
+      cond do
+        is_struct(date_a, Date) and is_nil(date_b) ->
```

Comment:
```suggestion
        is_nil(date_a) or is_nil(date_b) ->
```

I would suggest changing the check to this so that we can match nils in both positions